src/amo/components/SearchResult/index.js

```diff
@@ -55,16 +55,41 @@ export class SearchResultBase extends React.Component<InternalProps> {
 
     let imageURL = iconURL;
 
+    const imageAtts = {};
+
     if (isTheme) {
+      const hasPreviews = addon.previews.length;
+      // Since only newly created static themes will have more than one preview
+      // we will set up a fallback for now.
+      const previewFallback = hasPreviews && addon.previews[0];
+      const previewSearch = hasPreviews && addon.previews[1];
```

Comment:
It looks like flow is not happy unless I do something like this:

```javascript
const previewFallback = addon && addon.previews && addon.previews[0];
const previewSearch = addon && addon.previews && addon.previews[1];
```

basically, removing `hasPreviews`

I thought the above way above was cleaner and eliminating some redundancy. Am i missing something with flow :/?

Reply:
It's a limitation of Flow (a bug, too) -- it probably doesn't know that `hasPreviews` points to `addon.previews`. Sometimes you have to write verbose code to satisfy Flow. It's a trade-off.

Reply:
okay good to know... I wasn't sure if there was something I could have done differently with flow. thanks for looking and clarifying : )